erpnext/controllers/accounts_controller.py

```diff
@@ -2069,6 +2071,23 @@ def validate_quantity(child_item, d):
 		conv_fac_precision = child_item.precision("conversion_factor") or 2
 		qty_precision = child_item.precision("qty") or 2
 
+		user = frappe.session.user
+		if child_item.item_code != d.get("item_code"):
+			if ("Sales Manager" in frappe.get_roles(user)):
+				if parent_doctype in ['Sales Order', 'Purchase Order']:
+					qty_cond = child_item.delivered_qty if parent_doctype == 'Sales Order' else child_item.received_qty
+					if (child_item.billed_amt > 0 or qty_cond > 0):
+						frappe.throw(_("Cannot update item: partial bill or partial delivery exists."))
+					else:
+						item_data = frappe.get_doc("Item", d.get("item_code"))
+						child_item.item_code = d.get("item_code")
+						child_item.item_name = item_data.item_name
+						child_item.description = item_data.description
+						child_item.stock_uom = item_data.stock_uom
+						child_item.uom = item_data.stock_uom
+			else:
+				frappe.throw(_("User must be Sales Manager to update Item Code."))
```

Comment:
- We do not hardcode roles like this in code. Only Submit permission should be checked IMO since this is updating a submitted document. 
- Also same code runs for the purchase cycle where the Sales manager role doesn't make sense.